sisl/viz/plotly/plots/grid.py

```diff
@@ -512,6 +512,8 @@ def _set_data(self, axes, nsc, interp, trace_name, transforms, represent, cut_va
         elif represent.endswith("phase"):
             values = np.angle(grid.grid, deg=represent.startswith("deg"))
```

Comment:
What happens if `represent` isn't any of these?

Reply:
Good point, the obvious thing to do here would be
```
else:
    raise ValueError()
```
but since parameters are defined with input fields and input fields parse values before passing them to the plot I think it would be a good idea to raise the error when parsing, to avoid unnecessary calculations (the code would run until here, which might take a while, even though it's clearly deemed to fail). 

Also, raising the error on parse would make life much more easy for GUIs to know if there's a problem with a specific setting, for example, as they could check in real time if the setting is valid or not before trying to update the plot.

By the way, you have made me realise that there should probably be a new error type, something like `SislVizSettingValueError` for the same reason.